File: 7/prime_number_of_set_bits.go
package main

import (
	"fmt"
	"strconv"
)

func contains(arr []int, num int) bool {
	for i := 0; i < len(arr); i++ {
		if arr[i] == num {
			return true
		}
	}
	return false
}

func isPrime(num int) bool {
	if contains([]int{2, 3, 5, 7, 11, 13, 17, 19}, num) {
		return true
	}
	return false
}

func countPrimeSetBits(L int, R int) int {
	res := 0
	for num := L; num <= R; num++ {
		bin := strconv.FormatInt(int64(num), 2)

		count := 0
		for _, rune := range bin {
			if rune == '1' {
				count++
			}
		}

		if isPrime(count) {
			res++
		}
	}
	return res
}

func main() {
	fmt.Println(countPrimeSetBits(6, 10))
}

File: 7/prime_number_of_set_bits.py
class Solution:
    def countPrimeSetBits(self, L: int, R: int) -> int:
        def is_prime(num):
            return True if num in [2,3,5,7,11,13,17,19] else False
        
        res = 0
        for num in range(L, R+1):
            if is_prime(sum([1 if ch == "1" else 0 for ch in "{0:b}".format(num)])):
                res += 1
        
        return res
            
